### ISSUE TITLE
Preprocessing: standardize

### ISSUE THREAD

**Issue by anyaflya** [Peterande/D-FINE#313]
Hello, 

After going through the preprocessing of the data, I was wondering: why is there no z-score normalization? I see that the data is scaled between 0 and 1, through the transform `ConvertPILImage`. I also see that the `Normalize` function is available in the `_transform.py` file, while it is not used. 

I would be grateful to receive some insights why no z-score normalization / standardization is applied during the preprocessing step, or if I missed anything :) 

Best

**Comment by SebastianJanampa** [comment]
Hi @anyaflya 

You're right, the data is scaled between 0 and 1 using the ConverPILImage function with `scale=True` https://github.com/Peterande/D-FINE/blob/d6694750683b0c7e9f523ba6953d16f112a376ae/configs/dfine/include/dataloader.yml#L13

I'm not the author of the repo, but I think that using the `Normalize` function would affect the performance of the model since the `std` and `mean` drastically change between datasets.

**Comment by anyaflya** [comment]
Hello @SebastianJanampa ,

Thanks for your answer. 

Yes normalizing the datasets would negatively affect the performance of the model, if the datasets are different. If we assume the training dataset is similar to the test one, it is generally better to standardize the datasets (with the mean and std of the training dataset). I could see this preprocessing in other works, like [YOLOv11](https://docs.ultralytics.com/guides/preprocessing_annotated_data/#normalizing-pixel-values), which makes me wonder why standardization is not done here.

**Comment by SebastianJanampa** [comment]
YOLO offers `z-normalization` only for the classification task. Check this [link](https://github.com/ultralytics/ultralytics/blob/0fa42fbd7d7bd1535ebf627afff0385dea276147/ultralytics/data/augment.py#L22) and you will notice that `DEFAULT_MEAN` and `DEFAULT_STD` are only used for classification. You can also take a look at this [discussion](https://github.com/ultralytics/ultralytics/issues/10111). So, the quick answer is **NO**. YOLOv11 does not apply `z-normalization` during detection tasks.

**Comment by anyaflya** [comment]
thanks for the two links, I checked them out and I see that the `z-normalization` is only done for image classification. 

Would you have a longer answer as to why no `z-normalization` is performed for object detection @SebastianJanampa ?

**Comment by SebastianJanampa** [comment]
I think the main reason is speed.  D-FINE, as well as YOLO, focuses on fast inference. Using `z-normalization` will add the subtraction operation (division is also in scaling), which would add additional computation. Now, these models are already really fast, so the question is, would this significantly affect your requirements?  If you have the resources to train D-FINE using `z-normalization`, then do it. I think it's worth it, but it may not significantly increase the metrics. 

Now, in terms of code. Since scaling between 0 and 1 is a popular operation, the code for this process must be highly optimized. And `z-normalization` code may not be as fast, and the `scaling` operation. 
If you look at the inference code provided by the authors of this repo, you will notice that they use the `T.ToTensor` function, which is optimized and maintained by pytorch. They do not use the `ConvertPILImage` function used during training.
 https://github.com/Peterande/D-FINE/blob/d6694750683b0c7e9f523ba6953d16f112a376ae/tools/inference/torch_inf.py#L44-L49

I would also like to note that `z-normalization` is applied by object detectors like [DINO](https://github.com/IDEA-Research/DINO/blob/d84a491d41898b3befd8294d1cf2614661fc0953/datasets/coco.py#L456). Specifically, they utilize the `T.ToTensor` function followed by the `T.Normalize` function, which affects the inference time. `DINO` produces better results when it is pre-trained on the `Object365` (>60mAP), but those checkpoints are not and will not be released 😢 . Additionally, it requires more complex installations since you have to compile CUDA operators (you can get rid of them, but the model will become slower). If you wanna give them a look, you can check the [detrex](https://github.com/IDEA-Research/detrex/tree/main) library, where you will find many DETR-based models (it does not have any real-time detr models tho). FYI, detrx is not longer mantained

**Comment by anyaflya** [comment]
Thanks a lot for your detailed answer @SebastianJanampa ! I really appreciate the time you put in these explanations. It makes a lot more sense now. 

Inference time is not a requirement for my project. I think I'll add a scaling during inference, and compared the performance results. 

Cheers!
